nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/CaptureChangeMySQL.java

```diff
@@ -0,0 +1,928 @@
+            .required(true)
+            .expressionLanguageSupported(false)
+            .addValidator(StandardValidators.HOSTNAME_PORT_LIST_VALIDATOR)
+            .expressionLanguageSupported(true)
+            .build();
+
+    public static final PropertyDescriptor DRIVER_NAME = new PropertyDescriptor.Builder()
+            .name("capture-change-mysql-driver-class")
+            .displayName("MySQL Driver Class Name")
+            .description("The class name of the MySQL database driver class")
+            .defaultValue("com.mysql.jdbc.Driver")
+            .required(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .expressionLanguageSupported(true)
+            .build();
+
+    public static final PropertyDescriptor DRIVER_LOCATION = new PropertyDescriptor.Builder()
+            .name("capture-change-mysql-driver-locations")
+            .displayName("MySQL Driver Location(s)")
+            .description("Comma-separated list of files/folders and/or URLs containing the MySQL driver JAR and its dependencies (if any). "
+                    + "For example '/var/tmp/mysql-connector-java-5.1.38-bin.jar'")
+            .defaultValue(null)
+            .required(false)
+            .addValidator(StandardValidators.createListValidator(true, true, StandardValidators.createURLorFileValidator()))
+            .expressionLanguageSupported(true)
+            .build();
+
+    public static final PropertyDescriptor USERNAME = new PropertyDescriptor.Builder()
+            .name("capture-change-mysql-username")
+            .displayName("Username")
+            .description("Username to access the MySQL cluster")
+            .required(false)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .expressionLanguageSupported(true)
+            .build();
+
+    public static final PropertyDescriptor PASSWORD = new PropertyDescriptor.Builder()
+            .name("capture-change-mysql-password")
+            .displayName("Password")
+            .description("Password to access the MySQL cluster")
+            .required(false)
+            .sensitive(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .expressionLanguageSupported(true)
+            .build();
+
+    public static final PropertyDescriptor DIST_CACHE_CLIENT = new PropertyDescriptor.Builder()
+            .name("capture-change-mysql-dist-map-cache-client")
+            .displayName("Distributed Map Cache Client")
+            .description("Identifies a Distributed Map Cache Client controller service to be used for keeping information about the various tables, columns, etc. "
+                    + "needed by the processor. If a client is not specified, the generated events will not include column type or name information.")
+            .identifiesControllerService(DistributedMapCacheClient.class)
+            .required(false)
```

Comment:
I'm not sure of any use case where you wouldn't want table or column names, but I didn't want to prevent it just in case. If that isn't necessary, then I agree we should make it required.

Using a distributed cache vs internal variables was mostly to allow storage of possibly thousands of tables with hundreds of columns apiece. Certainly State Management was not the right solution there, but for performance I thought it might help to allow for persistent storage, for example if you restart NiFi you will lose internal state, but with a persistent cache you might have it "hydrated" when NiFi starts up. If it's overly complicated for the use case then I'm fine with an internal Map or something.